arraycontext/impl/pytato.py

```diff
@@ -0,0 +1,371 @@
+"""
+.. currentmodule:: arraycontext
```

Comment:
This should have some basic tests in place. Quite possibly, the existing tests for the pyopencl namespace should be broadened to include this array context. This could probably be done by making a new, modified/parametrized (?) version of the `actx_factory` fixture.

Reply:
Modified the test generator to also run the same tests for the PytatoArrayContext. Is this what you had in mind (minus the function argument)?

Reply:
Current test output:

```
FAILED test/test_arraycontext.py::test_array_context_np_workalike[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-sin-1]
FAILED test/test_arraycontext.py::test_array_context_np_workalike[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-exp-1]
FAILED test/test_arraycontext.py::test_array_context_np_workalike[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-arctan2-2]
FAILED test/test_arraycontext.py::test_array_context_np_workalike[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-minimum-2]
FAILED test/test_arraycontext.py::test_array_context_np_workalike[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-maximum-2]
FAILED test/test_arraycontext.py::test_array_context_np_workalike[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-where-3]
FAILED test/test_arraycontext.py::test_array_context_np_workalike[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-conj-1]
FAILED test/test_arraycontext.py::test_array_context_np_like[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-ones_like-1]
FAILED test/test_arraycontext.py::test_actx_stack[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>]
FAILED test/test_arraycontext.py::test_dof_array_arithmetic_same_as_numpy[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>]
FAILED test/test_arraycontext.py::test_dof_array_reductions_same_as_numpy[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>]
FAILED test/test_arraycontext.py::test_array_context_einsum_array_manipulation[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-ij->ij]
FAILED test/test_arraycontext.py::test_array_context_einsum_array_manipulation[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-ij->ji]
FAILED test/test_arraycontext.py::test_array_context_einsum_array_manipulation[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-ii->i]
FAILED test/test_arraycontext.py::test_array_context_einsum_array_matmatprods[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-ij,ij->ij]
FAILED test/test_arraycontext.py::test_array_context_einsum_array_matmatprods[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-ij,ji->ij]
FAILED test/test_arraycontext.py::test_array_context_einsum_array_matmatprods[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-ij,kj->ik]
FAILED test/test_arraycontext.py::test_array_context_einsum_array_tripleprod[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-im,mj,k->ijk]
FAILED test/test_arraycontext.py::test_container_multimap[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>]
FAILED test/test_arraycontext.py::test_container_arithmetic[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>]
FAILED test/test_arraycontext.py::test_container_norm[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-2]
FAILED test/test_arraycontext.py::test_container_norm[<Pytato array context factory for <pyopencl.Device 'pthread' on 'Portable Computing Language'>-inf]
============================== 22 failed, 32 passed, 501 warnings in 8.11s ==============================
```